src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            last = new Node<>(null, value, null);
+            first = last;
+        } else {
+            Node<T> addElement = new Node<>(last, value, null);
+            last.next = addElement;
+            last = addElement;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        indexOutOfBoundsException(index);
+        if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, first);
+            first.prev = newNode;
+            first = newNode;
+        } else {
+            Node<T> current = find(index);
+            Node<T> addIndex = new Node<>(null, value, current);
+            current.prev.next = addIndex;
+            current.prev = addIndex;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexOutOfBoundsException(index);
+        return find(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexOutOfBoundsException(index);
+        Node<T> current = find(index);
+        T oldValue = current.item;
+        current.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexOutOfBoundsException(index);
+        Node<T> removed = find(index);
+        final T result = removed.item;
+        if (removed.next == null) {
+            last = removed.prev;
+        } else {
+            removed.next.prev = removed.prev;
+        }
+        if (removed.prev == null) {
+            first = removed.next;
+        } else {
+            removed.prev.next = removed.next;
+        }
+        size--;
```

Comment:
Move this logic to separate functuin(e.g. unlink)
```suggestion
        unlink(removed);
```